src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+                baseQualities.put(currentBase, currentBaseQualityList);
+            }
+            currentBaseQualityList.add(new BaseQuality(
+                    errorProbabilityAdjustmentFactor * QualityUtils.qualToErrorProb(rec.getBaseQuality()),
+                    errorProbabilityAdjustmentFactor * QualityUtils.qualToErrorProb(rec.getRecord().getMappingQuality()))
+            );
+        }
+
+        return baseQualities;
+    }
+
+    /**
+     * Returns base-pair counts at a given locus.
+     * @param locus locus
+     * @return      base-pair counts
+     */
+    static Nucleotide.Counter getPileupBaseCounts(final SamLocusIterator.LocusInfo locus) {
+        final Nucleotide.Counter result = new Nucleotide.Counter();
+        for (final SamLocusIterator.RecordAndOffset rec : locus.getRecordAndPositions()) {
+            result.add(rec.getReadBase());
+        }
+        return result;
+    }
+
+    /**
+     * get base counts map from base quality map
+     * @param baseQualities map from bases to list of {@link BaseQuality}
+     * @return base counts map
+     */
+    private Map<Nucleotide, Integer> getBaseCountsFromBaseQualities(final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities) {
+        Map<Nucleotide, Integer> baseCounts = new HashMap<>();
+        for (Nucleotide base : BASES) {
+            if (baseQualities.containsKey(base)) {
+                baseCounts.put(base, baseQualities.get(base).size());
+            } else {
+                baseCounts.put(base, 0);
+            }
+        }
+        return baseCounts;
+    }
+
+    /**
+     * Infer the alt allele base from the pileup. We choose the base with the highest frequency
+     * that is not Ref (this coincides with the maximum likelihood estimation).
+     *
+     * * Remark: while this is definitely not the perfect way to do it, it has no serious pitfalls:
+     *
+     * - if the input is balanced between betweet ref and alt, it correctly chooses alt.
+     * - if the input is all ref, then alt is chosen randomly, but it's OK since there are no alt counts
+     *   in the pileup anyway.
+     * - if the input is all alt, we are good.
+     * - if the input in all ref + a few errors, then alt will be choosen as the most frequent erroneous
+     *   base; again, this is OK because Hom likelihood is far greater than Het likelihood in this case
+     * - if the input is all alt + a few errors, we are good.
+
+     * @param baseQualities map from bases to error probabilities
+     * @param refBase the ref allele base
+     * @return the likely alt allele
+     */
+    @VisibleForTesting
+    public static Nucleotide inferAltFromPileup(final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities,
+                                                final Nucleotide refBase) {
+
+        /* sort the bases in the descending order by their frequency */
+        Nucleotide[] bases = {Nucleotide.A, Nucleotide.C, Nucleotide.G, Nucleotide.T};
```

Comment:
Yes, but `bases` is to be sorted based on frequency in the pileup and is not constant. Now I initialize it to to `BASES` rather than declaring ACTG explicitly here again.